python_modules/dagster-graphql/dagster_graphql/schema/roots/query.py

```diff
@@ -281,13 +288,22 @@ class Meta:
 
     instigationStateOrError = graphene.Field(
         graphene.NonNull(GrapheneInstigationStateOrError),
-        instigationSelector=graphene.NonNull(GrapheneInstigationSelector),
+        instigationSelector=graphene.Argument(GrapheneInstigationSelector),
+        id=graphene.Argument(graphene.String),
```

Comment:
Can this take an array of Ids instead so that I can use our batching abstraction :)

Reply:
i can on the "states" resolver -I need to add the database look impl for many by id so will be a stacked PR. Thought maybe converting "SingleXQuery` in place would be the move but I can see wanting to move to batching while we are changing things

Reply:
I don't mind keeping individual queries. It's likely faster since theres more parallelization, but also slight more taxing on the server. gcpu vs ux latency trade off